src/app/leaves/add/components/LeaveRequestFormBuilder.tsx

```diff
@@ -0,0 +1,262 @@
+                      </Button>
+                    </FormControl>
+                  </PopoverTrigger>
+                  <PopoverContent className="w-auto p-0" align="start">
+                    <Calendar
+                      mode="single"
+                      selected={field.value}
+                      onSelect={field.onChange}
+                      initialFocus
+                    />
+                  </PopoverContent>
+                </Popover>
+                <FormMessage />
+              </FormItem>
+            )}
+          />
+          <FormField
+            control={form.control}
+            name="endDate"
+            render={({ field }) => (
+              <FormItem className="flex flex-col">
+                <FormLabel>End Date</FormLabel>
+                <Popover>
+                  <PopoverTrigger asChild>
+                    <FormControl>
+                      <Button
+                        data-testid="calendar-end-date-button"
+                        variant={"outline"}
+                        className={cn(
+                          `pl-3 text-left font-normal ${
+                            errors.endDate
+                              ? "border-destructive"
+                              : "border-input"
+                          }`,
+                          !field.value && "text-muted-foreground"
+                        )}
+                      >
+                        {field.value ? (
+                          format(field.value, "PPP")
+                        ) : (
+                          <span>Pick a date</span>
+                        )}
+                        <CalendarIcon className="ml-auto h-4 w-4 opacity-50" />
+                      </Button>
+                    </FormControl>
+                  </PopoverTrigger>
+                  <PopoverContent
+                    data-testid="popover"
+                    className="w-auto p-0"
+                    align="start"
+                  >
+                    <Calendar
+                      data-testid="calendar"
+                      mode="single"
+                      selected={field.value}
+                      onSelect={field.onChange}
+                      initialFocus
+                    />
+                  </PopoverContent>
+                </Popover>
+                <FormMessage />
+              </FormItem>
+            )}
+          />
+
+          <FormField
+            control={form.control}
+            name="reasonForLeave"
+            render={({ field }) => (
+              <FormItem>
+                <FormLabel>Reason for Leave</FormLabel>
+                <FormControl>
+                  <Textarea
+                    placeholder="Please provide a brief description of your leave reason."
+                    data-testid="leaveReason"
+                    className={`resize-none ${
+                      errors.reasonForLeave
```

Comment:
can we use hasError prop of this TextArea input for showing error

Reply:
@Mubashar1009  I'm not sure about the `hasError` initially, but I've checked this prop now and fixed the issue.